### PR TITLE
Execute functional testing locally

### PR DESCRIPTION
**Pull request by FenrirUnbound**
Pain points so far:
* Requires a Github token to perform Github operations.
* Requires a valid JWT. It needs it to create a pipeline via Screwdriver API
* Waiting for certain events (e.g., build to queue up or abort) isn't the cleanest solution. The tests wait a certain period of time for events to propagate through the system; it could be smarter but it makes me question my approach in the first place.

How to run:
screen shot 2016-08-24 at 6 06 22 pm

### CODE REVIEW

File: functional [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
```
Comment (stjohnjohnson) [inline functional]: you COULD replace all of this with some global setup feature of cucumber.

File: README.md [hunk 2]
```diff
@@ -82,14 +82,18 @@ npm test
 
 ### Functional tests
 
-First start the demo server:
-```bash
-npm start
+Create a `.func_config.json` file with the following content:
```
Comment (nkatzman) [inline README.md]: Should this be telling you to `updated`?

File: features/step_definitions/git-flow.js [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+
+    this.Then(/^any existing builds should be stopped$/, { timeout: 15 * 1000 }, () =>
+        promiseToWait(5)
+        .then(() => waitForBuild(this.instance, this.sha))
+        .then((data) => {
+            const build = data[0];
+
+            Assert.oneOf(build.status, ['ABORTED', 'SUCCESS']);
+
+            return {};
```
Comment (tkyi) [inline features/step_definitions/git-flow.js]: why do we return empty object here?

File: README.md [hunk 4]
```diff
@@ -82,14 +82,18 @@ npm test
 
 ### Functional tests
 
-First start the demo server:
-```bash
-npm start
+Update `.func_config.json` with your own username, github_token, and jwt:
```
Comment (stjohnjohnson) [inline README.md]: I'm not super happy about this as people can accidentally commit their info

**Comment by stjohnjohnson** [comment]
Otherwise, 👍  seems reasonable.

**Comment by tkyi** [comment]
👍 looks like a good workaround for now

**Pull request merged**